spec/architecture.md

```diff
@@ -0,0 +1,137 @@
+    porterAdmin((Porter Admin))--deploys---->BondFactory
+    Bond["Bond (tokenImplementation)"]
+```
+### 2. Issuer Contacts Porter Finance
+   - Porter Finance evaluates Issuer's credit worthyness
+   - Porter Finance creates parameters for the issuance
+     - Amount of ERC20 token backing each share (Collateral Token)
+     - Amount of collateral each share would convert into
+     - Maturity date 
+     - Total issuance size
+     - The ERC20 token the Bond is denominated in (Payment Token)
+   - Porter Finance adds Issuer to allow list
+   - Porter Finance adds Collateral Token & Payment Token to allow list
+   
+### 3. Issuer Creates a Bond
+After the issuance has been approved, the issuer will be on the allowed issuer list and their intended collateral token and payment token will be added to the allowed token list. The Bonds are created (minted) at the same time that the collateral is deposited. After a Bond is created, the issuer can sell the Bond through the [Porter Finance App](https://app.porter.finance) (we use [Gnosis' Batch Auction](https://github.com/gnosis/ido-contracts)).
+
+These are sold as [Zero Coupon Bonds](https://docs.porter.finance/portal/financial-concepts/zero-coupon-bonds). Another party will purchase these bond shares at a discount by paying with the Payment Token. The purchasers are known as **bond holders**. At maturity, the bond holders can redeem their bond shares at 1 to 1 for Payment Tokens.
+
+```mermaid
+  flowchart LR
+    issuer((Issuer))
+    subgraph Bond Factory
+    issuerList{Is issuer allow list enabled?}
+    tokenList{Is token allow list enabled?}
+    hasIssuerRole{Has Issuer Role?}
+    hasTokenRole{Has Token Role?}
+    createBond[Create Bond]
+    deposit{Has enough collateral?}
+    NewBond[Deposit & Create new Bond]
+    end
+    
+    issuer--with configuration-->createBond
+    createBond-->issuerList
+    issuerList--yes-->hasIssuerRole
+    issuerList--no-->tokenList
+    tokenList--yes-->hasTokenRole
+    tokenList--no-->deposit
+    hasIssuerRole--yes-->tokenList
+    hasTokenRole--yes-->deposit
+    hasIssuerRole & hasTokenRole--no-->End
+    deposit--yes-->NewBond
+    deposit--no-->End
+```
+
+## Bond States
+Issuers and Bond Holders can both interact with the bond in a few ways. For a simpler view, look at the [state transition diagram](/spec/stateMachine.md). Since the Bond is an ERC20 token, the bond shares may be **`transferred`** at any time by any users.
+
+### Issuer States
+Issuers can also be thought of as bond holders. In addition to the actions taken by bond holders, Issuers have the ability to **`pay`**, **`withdrawExcessCollateral`**, **`withdrawExcessPayment`** and **`sweep`**. These can be called at any time to affect the Bond.
+
+#### Decrease of collateral requirement
+
+There are three ways in which the collateral requirement can decrease. At any time, the Issuer may `withdrawExcessCollateral` from the contract in order to reclaim some of their collateral.
```

Comment:
> There are three ways in which the collateral requirement can decrease. At any time, the Issuer may `withdrawExcessCollateral` from the contract in order to reclaim some of their collateral.

Instead of >  At any time, the Issuer may `withdrawExcessCollateral` from the contract in order to reclaim some of their collateral.



I'd prob say something like "If there is more collateral in the contract that required, the Issuer may withdrawExcessCollateral from the contract in order to reclaim some of their collateral